File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        map<int,bool> m;
        m[prices[0]] = true;
        int maxProfit = 0;
        for(int i=1;i<prices.size();i++){
            int profit = prices[i] - m.begin()->first ;
            maxProfit = max(profit,maxProfit);
            m[prices[i]] = true;
        }
        return maxProfit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.php
class Solution {

    /**
     * @param Integer[] $prices
     * @return Integer
     */
    function maxProfit($prices) {
        $miprice = $prices[0];
        $profit = 0;

        foreach($prices as $price){
            $profit = max($profit, ($price - $miprice));
            $miprice = min($miprice,$price);
        }
        return $profit;
        
    }
}